api/shapes/loadouts.ts

```diff
@@ -118,28 +134,43 @@ export interface LoadoutParameters {
    * which constrains the items that can be in the loadout.
    */
   query?: string;
+
   /**
    * When generating the loadout, did we assume all items were at their
    * masterworked stats, or did we use their current stats?
    *
-   * @deprecated use upgradeSpendTier
+   * @deprecated use assumeArmorMasterworked
    */
   assumeMasterworked?: boolean;
 
   /**
    * What upgrades are the user willing to shell out for?
+   *
+   * @deprecated use assumeArmorMasterworked
    */
   upgradeSpendTier?: UpgradeSpendTier;
 
+  /**
+   * Whether armor of this type will have assumed materwork stats in the Loadout Optimizer.
+   */
+  assumeArmorMasterwork?: AssumeArmorMasterwork;
```

Comment:
Is undefined the same as no assume masterwork? Is it possible to just map these off the existing settings so we don't have to invalidate everyone's saved loadouts?

Reply:
That was the intent yes. After the issues with deprecating enums last time I was aiming to try and make this version a little more resilient. Where would you want to do the mapping? In a SQL script or an explicit mapping?

Reply:
I meant can we keep the settings we have and just map them back and forth to the new switches in the UI rather than losing all the new data.

Reply:
It won't be a 1-1 mapping. There was behaviour in the old settings that can be replicated with these toggles but one setting could essentially be split across both. For example the "UpgradeSpendTier.AscendantShardsNotMasterwork" was essentially 
`AssumeArmorMasterwork.All + LockArmorEnergyType.Masterworked`

Reply:
Gotcha. I guess I'm just bummed that all saved data will be broken.

Reply:
Yeah I am too, which is why I am open to have some kind of mapping/SQL solution that would at least mitigate the issue for a while. For example, if we have a mapping solution (either in here or DIM) we could automatically convert loadouts that come in with the old options and then save them in the background. Alternative is populating the data in a SQL script, I haven't written SQL in a couple of years now is the only concern and I don't remember ever doing DB migration scripts for data with JSON field updates.

Reply:
Either of those is fine, but if we can do that mapping, do we need to transform the data at all, or can we just always do it in the UI?

Reply:
We could do it in the UI. The question is do we want to actively save the mapped version on load or do we want to only save it if the user saves it?

I'm inclined to set this up, run it for a couple of months, and then pull it all out and only have the new version. Anyone that hasn't played in that time we lose the settings.

Reply:
One thought I had was that doing it in the api as a temporary measure would reduce chatter over the wire. 

I imagine 
1. A user opens the app and we make the call to get their loadouts
2. We upgrade each loadout
3. Save back to the DB
4. Send the response with the loadouts.

Reply:
We could do that. If we have such a function the easiest thing to do would just be to read and write the old format to the API from DIM's code, and then we could slowly start moving it over.